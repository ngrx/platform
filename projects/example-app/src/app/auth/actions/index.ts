import { AuthActions, AuthActionTypes } from './auth.actions';
import { AuthApiActions, AuthApiActionTypes } from './auth-api.actions';
import { LoginPageActions, LoginPageActionTypes } from './login-page.actions';

export {
  AuthActions,
  AuthActionTypes,
  AuthApiActions,
  AuthApiActionTypes,
  LoginPageActions,
  LoginPageActionTypes,
};
