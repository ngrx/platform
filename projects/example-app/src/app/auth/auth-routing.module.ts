import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { LoginPageComponent } from './containers/login-page.component';

const routes: Routes = [{ path: 'login', component: LoginPageComponent }];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class AuthRoutingModule {}
