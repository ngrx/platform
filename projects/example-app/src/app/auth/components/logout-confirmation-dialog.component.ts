import { Component } from '@angular/core';

/**
 * The dialog will close with true if user clicks the ok button,
 * otherwise it will close with undefined.
 */
@Component({
  template: `
    <h2 mat-dialog-title>Logout</h2>
    <mat-dialog-content>Are you sure you want to logout?</mat-dialog-content>
    <mat-dialog-actions>
      <button mat-button [mat-dialog-close]="false">Cancel</button>
      <button mat-button [mat-dialog-close]="true">OK</button>
    </mat-dialog-actions>
  `,
  styles: [
    `
      :host {
        display: block;
        width: 100%;
        max-width: 300px;
      }

      mat-dialog-actions {
        display: flex;
        justify-content: flex-end;
      }

      [mat-button] {
        padding: 0;
      }
    `,
  ],
})
export class LogoutConfirmationDialogComponent {}
