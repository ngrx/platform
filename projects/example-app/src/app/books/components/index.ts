export * from './book-authors.component';
export * from './book-detail.component';
export * from './book-preview.component';
export * from './book-preview-list.component';
export * from './book-search.component';
