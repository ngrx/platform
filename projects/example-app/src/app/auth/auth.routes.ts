import { Routes } from '@angular/router';
import { LoginPageComponent } from '@example-app/auth/containers';

export const AUTH_ROUTES: Routes = [
  {
    path: '',
    component: LoginPageComponent,
    data: { title: 'Login' },
  },
];
