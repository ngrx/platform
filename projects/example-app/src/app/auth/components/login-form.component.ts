import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { FormGroup, FormControl } from '@angular/forms';
import { Credentials } from '@example-app/auth/models';

@Component({
  selector: 'bc-login-form',
  template: `
    <mat-card>
      <mat-card-title>Login</mat-card-title>
      <mat-card-content>
        <form [formGroup]="form" (ngSubmit)="submit()">
          <p>
            <mat-form-field>
              <input
                type="text"
                matInput
                placeholder="Username"
                formControlName="username"
              />
            </mat-form-field>
          </p>

          <p>
            <mat-form-field>
              <input
                type="password"
                matInput
                placeholder="Password"
                formControlName="password"
              />
            </mat-form-field>
          </p>

          <p *ngIf="errorMessage" class="loginError">
            {{ errorMessage }}
          </p>

          <p class="loginButtons">
            <button type="submit" mat-button>Login</button>
          </p>
        </form>
      </mat-card-content>
    </mat-card>
  `,
  styles: [
    `
      :host {
        display: flex;
        justify-content: center;
        margin: 72px 0;
      }

      .mat-form-field {
        width: 100%;
        min-width: 300px;
      }

      mat-card-title,
      mat-card-content {
        display: flex;
        justify-content: center;
      }

      .loginError {
        padding: 16px;
        width: 300px;
        color: white;
        background-color: red;
      }

      .loginButtons {
        display: flex;
        flex-direction: row;
        justify-content: flex-end;
      }
    `,
  ],
})
export class LoginFormComponent {
  @Input()
  set pending(isPending: boolean) {
    if (isPending) {
      this.form.disable();
    } else {
      this.form.enable();
    }
  }

  @Input() errorMessage!: string | null;

  @Output() submitted = new EventEmitter<Credentials>();

  form: FormGroup = new FormGroup({
    username: new FormControl('ngrx'),
    password: new FormControl(''),
  });

  submit() {
    if (this.form.valid) {
      this.submitted.emit(this.form.value);
    }
  }
}
