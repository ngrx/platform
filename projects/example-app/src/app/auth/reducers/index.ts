import {
  createSelector,
  createFeatureSelector,
  Action,
  combineReducers,
} from '@ngrx/store';
import * as fromRoot from '@example-app/reducers';
import * as fromAuth from '@example-app/auth/reducers/auth.reducer';
import * as fromLoginPage from '@example-app/auth/reducers/login-page.reducer';

export const authFeatureKey = 'auth';

export interface AuthState {
  [fromAuth.statusFeatureKey]: fromAuth.State;
  [fromLoginPage.loginPageFeatureKey]: fromLoginPage.State;
}

export interface State extends fromRoot.State {
  [authFeatureKey]: AuthState;
}

export function reducers(state: AuthState | undefined, action: Action) {
  return combineReducers({
    [fromAuth.statusFeatureKey]: fromAuth.reducer,
    [fromLoginPage.loginPageFeatureKey]: fromLoginPage.reducer,
  })(state, action);
}

export const selectAuthState = createFeatureSelector<State, AuthState>(
  authFeatureKey
);

export const selectAuthStatusState = createSelector(
  selectAuthState,
  (state) => state.status
);
export const selectUser = createSelector(
  selectAuthStatusState,
  fromAuth.getUser
);
export const selectLoggedIn = createSelector(selectUser, (user) => !!user);

export const selectLoginPageState = createSelector(
  selectAuthState,
  (state) => state.loginPage
);
export const selectLoginPageError = createSelector(
  selectLoginPageState,
  fromLoginPage.getError
);
export const selectLoginPagePending = createSelector(
  selectLoginPageState,
  fromLoginPage.getPending
);
