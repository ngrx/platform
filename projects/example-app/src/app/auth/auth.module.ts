import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { LoginPageComponent } from '@example-app/auth/containers';
import {
  LoginFormComponent,
  LogoutConfirmationDialogComponent,
} from '@example-app/auth/components';

import { AuthEffects } from '@example-app/auth/effects';
import * as fromAuth from '@example-app/auth/reducers';
import { MaterialModule } from '@example-app/material';
import { AuthRoutingModule } from './auth-routing.module';

export const COMPONENTS = [
  LoginPageComponent,
  LoginFormComponent,
  LogoutConfirmationDialogComponent,
];

@NgModule({
  imports: [
    CommonModule,
    ReactiveFormsModule,
    MaterialModule,
    AuthRoutingModule,
    StoreModule.forFeature({
      name: fromAuth.authFeatureKey,
      reducer: fromAuth.reducers,
    }),
    EffectsModule.forFeature([AuthEffects]),
  ],
  declarations: COMPONENTS,
  entryComponents: [LogoutConfirmationDialogComponent],
})
export class AuthModule {}
