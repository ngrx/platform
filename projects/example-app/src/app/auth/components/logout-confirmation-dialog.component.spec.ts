import { ComponentFixture, TestBed } from '@angular/core/testing';
import { LogoutConfirmationDialogComponent } from '@example-app/auth/components';
import { MaterialModule } from '@example-app/material';

describe('Logout Confirmation Dialog', () => {
  let fixture: ComponentFixture<LogoutConfirmationDialogComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [MaterialModule],
      declarations: [LogoutConfirmationDialogComponent],
    });

    fixture = TestBed.createComponent(LogoutConfirmationDialogComponent);
  });

  it('should compile', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
