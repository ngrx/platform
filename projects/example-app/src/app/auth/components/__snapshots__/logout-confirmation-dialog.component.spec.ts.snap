// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Confirmation Dialog should compile 1`] = `
<ng-component>
  <h2
    class="mat-dialog-title"
    mat-dialog-title=""
  >
    Logout
  </h2><mat-dialog-content
    class="mat-dialog-content"
  >
    Are you sure you want to logout?
  </mat-dialog-content><mat-dialog-actions
    class="mat-dialog-actions"
  >
    <button
      class="mat-focus-indicator mat-button mat-button-base"
      mat-button=""
      type="button"
    >
      <span
        class="mat-button-wrapper"
      >
        Cancel
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
    <button
      class="mat-focus-indicator mat-button mat-button-base"
      mat-button=""
      type="button"
    >
      <span
        class="mat-button-wrapper"
      >
        OK
      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </mat-dialog-actions>
</ng-component>
`;
